### ISSUE TITLE
EVFEVENT error

### ISSUE THREAD

**Issue by zkarj735** [IBM/ibmi-bob#133]
As executed from VSCode using 2.4.0.1
```
Current library: QGPL
Library list: QTEMP DAGSW DAGDTA QGPL TAATOOL IOMSDTA
Commands:
		OPT=*EVENTF BUILDLIB=DEMOTEST /QOpenSys/pkgs/bin/makei build

> /QOpenSys/pkgs/bin/make -k BUILDVARSMKPATH="/tmp/tmp7k9ssotr" -k BOB="/QOpenSys/pkgs/lib/bob" -f "/QOpenSys/pkgs/lib/bob/mk/Makefile" all
=== Creating DSPF [fmddspaltd.dspf]
/QOpenSys/pkgs/lib/bob/scripts/crtfrmstmf -f /home/ajenks/ibmi-alert-framework/fmddspaltd.dspf -o FMDDSPALTD -l DEMOTEST -c CRTDSPF -p "ENHDSP(*YES) RSTDSP(UNKNOWN_FILE_TYPE) DFRWRT(*YES) AUT(UNKNOWN_FILE_TYPE) OPTION(UNKNOWN_FILE_TYPE) TEXT()"
✕ Failed to create FMDDSPALTD.FILE!

CPFA0A9: Object not found.  Object is /QSYS.LIB/DEMOTEST.LIB/EVFEVENT.FILE/FMDDSPALTD.MBR.
CPFA097: Object not copied.  Object is /QSYS.LIB/DEMOTEST.LIB/EVFEVENT.FILE/FMDDSPALTD.MBR.
make: *** [/QOpenSys/pkgs/lib/bob/mk/Makefile:67: FMDDSPALTD.FILE] Error 255
=== Create panel group [FMHDSPALT]
/QOpenSys/pkgs/lib/bob/scripts/crtfrmstmf -f /home/ajenks/ibmi-alert-framework/fmhdspalt.pnlgrpsrc -o FMHDSPALT -l DEMOTEST -c CRTPNLGRP -p "AUT() OPTION(*EVENTF *SRC) TEXT()"
✓ FMHDSPALT.PNLGRP was created successfully!

make: Target 'all' not remade because of errors.
Objects:             1 failed 1 succeed 2 total
 > Failed objects:    FMDDSPALTD.FILE
Build Completed!

Fetching errors from QGPL/fmhdspalt.
```
I had this same issue with a PGM that I was attempting previously (which had its own issues discussed in #131). The same error appears running from the command line.

**Comment by edmundreinhardt** [comment]
The CRTDSPF  is attempting to compile FMDDSPALTD in the library DEMOTEST
The errors should be generated into an event file in the same library (DEMOTEST) and in the SRC-PF EVFEVENT and with the same name as the member being compiled .  This is standard behaviour of CRTDSPF if OPTION(*EVENTF) is specified.
Now the crtfrmstmf attempts to copy this member back to the .evfevent  directory in the IFS project root.   Code for IBM i will copy back the .evfevent files and populate the Problem Diagnostics based on the ERROR lines in teh EVFEVENT files.

**Comment by edmundreinhardt** [comment]
What makes me suspicious in your case is that it is fetching errore from QGPL according to the last line.

**Comment by worksofliam** [comment]
The current library needs to be set the library that is being built into for the errors to fetch correctly. This is because we're building from local.

**Comment by zkarj735** [comment]
It was mentioned in #129 that VSCode may not respect the current library, however I get the same errors running from the command line. I am specifying the current library per examples. It occurs to me if the current and build libraries need to be the same, then this should be abstracted away from the users, or at least clearly stated as a requirement and in examples.

iproj.json
```
{
    "version": "0.0.1",
    "description": "Alerting framework",
    "objlib": "&BUILDLIB",
    "curlib": "&BUILDLIB",
    "preUsrlibl": "&BUILDLIB"
}
```
Rules.mk
```
CMDs := DSPALT.CMD
DSPFs := FMDDSPALTD.FILE
MODULEs := FMRDSPALTD.MODULE
PGMs := FMRDSPALTD.PGM
PNLGRPs := FMHDSPALT.PNLGRP

DSPALT.CMD: dspalt.cmdsrc
DSPALT.CMD: private PGM := FMRALTSND
DSPALT.CMD: private HLPPNLGRP := FMHDSPALT
FMDDSPALTD.FILE: fmddspaltd.dspf
FMRDSPALTD.MODULE: fmrdspaltd.sqlrpgle FMDDSPALTD.FILE
FMRDSPALTD.PGM: fmrdspaltd.ilepgm FMRDSPALTD.MODULE
FMHDSPALT.PNLGRP: fmhdspalt.pnlgrpsrc
```
The command
```
 OPT=*EVENTF BUILDLIB=DEMOTEST /QOpenSys/pkgs/bin/makei build
```
The result
```
> /QOpenSys/pkgs/bin/make -k BUILDVARSMKPATH="/tmp/tmpqr5o0myu" -k BOB="/QOpenSys/pkgs/lib/bob" -f "/QOpenSys/pkgs/lib/bob/mk/Makefile" all
=== Creating DSPF [fmddspaltd.dspf]
/QOpenSys/pkgs/lib/bob/scripts/crtfrmstmf -f /home/ajenks/ibmi-alert-framework/fmddspaltd.dspf -o FMDDSPALTD -l DEMOTEST -c CRTDSPF -p "ENHDSP(*YES) RSTDSP(UNKNOWN_FILE_TYPE) DFRWRT(*YES) AUT(UNKNOWN_FILE_TYPE) OPTION(UNKNOWN_FILE_TYPE) TEXT()"
✕ Failed to create FMDDSPALTD.FILE!

CPFA0A9: Object not found.  Object is /QSYS.LIB/DEMOTEST.LIB/EVFEVENT.FILE/FMDDSPALTD.MBR.
CPFA097: Object not copied.  Object is /QSYS.LIB/DEMOTEST.LIB/EVFEVENT.FILE/FMDDSPALTD.MBR.
make: *** [/QOpenSys/pkgs/lib/bob/mk/Makefile:67: FMDDSPALTD.FILE] Error 255
=== Create panel group [FMHDSPALT]
/QOpenSys/pkgs/lib/bob/scripts/crtfrmstmf -f /home/ajenks/ibmi-alert-framework/fmhdspalt.pnlgrpsrc -o FMHDSPALT -l DEMOTEST -c CRTPNLGRP -p "AUT() OPTION(*EVENTF *SRC) TEXT()"
✕ Failed to create FMHDSPALT.PNLGRP!

make: Target 'all' not remade because of errors.
Objects:             2 failed 0 succeed 2 total
 > Failed objects:    FMDDSPALTD.FILE FMHDSPALT.PNLGRP
Build Completed!
```

**Comment by zkarj735** [comment]
Per #129 I made the change to using VSCode's current library as the build library. This is now reflected correctly in the output, however I still get the error.

```
Current library: DEMOTEST
Library list: QTEMP DAGSW DAGDTA QGPL TAATOOL IOMSDTA
Commands:
		err=*EVENTF BUILDLIB=DEMOTEST /QOpenSys/pkgs/bin/makei build

> /QOpenSys/pkgs/bin/make -k BUILDVARSMKPATH="/tmp/tmpdoegr7rz" -k BOB="/QOpenSys/pkgs/lib/bob" -f "/QOpenSys/pkgs/lib/bob/mk/Makefile" all
=== Creating DSPF [FMDDSPALTD.DSPF]
/QOpenSys/pkgs/lib/bob/scripts/crtfrmstmf -f /home/ajenks/ibmi-alert-framework/FMDDSPALTD.DSPF -o FMDDSPALTD -l DEMOTEST -c CRTDSPF -p "ENHDSP(*YES) RSTDSP(*YES) DFRWRT(*YES) AUT() OPTION(*EVENTF *SRC *LIST) TEXT()"
✕ Failed to create FMDDSPALTD.FILE!

CPFA0A9: Object not found.  Object is /QSYS.LIB/DEMOTEST.LIB/EVFEVENT.FILE/FMDDSPALTD.MBR.
CPFA097: Object not copied.  Object is /QSYS.LIB/DEMOTEST.LIB/EVFEVENT.FILE/FMDDSPALTD.MBR.
make: *** [/QOpenSys/pkgs/lib/bob/mk/Makefile:67: FMDDSPALTD.FILE] Error 255
=== Create panel group [FMHDSPALT]
/QOpenSys/pkgs/lib/bob/scripts/crtfrmstmf -f /home/ajenks/ibmi-alert-framework/FMHDSPALT.PNLGRPSRC -o FMHDSPALT -l DEMOTEST -c CRTPNLGRP -p "AUT() OPTION(*EVENTF *SRC) TEXT()"
✕ Failed to create FMHDSPALT.PNLGRP!

make: Target 'all' not remade because of errors.
Objects:             2 failed 0 succeed 2 total
 > Failed objects:    FMDDSPALTD.FILE FMHDSPALT.PNLGRP
Build Completed!

Fetching errors from DEMOTEST/fmhdspalt.
```
I do not get the error if just doing a `compile`, however that fails with what I believe is a letter case issue.
```
Current library: DEMOTEST
Library list: QTEMP DAGSW DAGDTA QGPL TAATOOL IOMSDTA
Commands:
		err=*EVENTF /QOpenSys/pkgs/bin/makei compile -f fmhdspalt.pnlgrpsrc -e BUILDLIB=DEMOTEST

Set variable <BUILDLIB> to 'DEMOTEST'
> /QOpenSys/pkgs/bin/make -k BUILDVARSMKPATH="/tmp/tmpuw5oylc9" -k BOB="/QOpenSys/pkgs/lib/bob" -f "/QOpenSys/pkgs/lib/bob/mk/Makefile" fmhdspalt.PNLGRP
make: *** No rule to make target 'fmhdspalt.PNLGRP'.
Objects:             0 failed 0 succeed 0 total
Build Completed!

Fetching errors from DEMOTEST/fmhdspalt.
```

**Comment by edmundreinhardt** [comment]
@zkarj735 is your code available on github for us to reproduce?

**Comment by zkarj735** [comment]
@edmundreinhardt it isn't currently, as it's in my company ADO. I could work to produce a sanitised version to put on Github, but wouldn't be able to test from there. But is this an issue with the code (which I know compiles in 5250-land) or more likely my environment/assumptions made by bob?

There is every chance my display file and panel group don't compile given the context, but I can't tell why they won't compile without access to the errors that EVFEVENT should provide. Unless I can see them somewhere in IBM i jobs?

**Comment by zkarj735** [comment]
@edmundreinhardt I have cloned the project up to Github in its current state. It's missing the actual files so will not run, but as they are accessed via SQL it should all compile. In any case, having it attempt the compile of all of the objects would be a start.

https://github.com/zkarj735/alert-framework

**Comment by edmundreinhardt** [comment]
Closed a duplicate of #129
